Account model

vulturana1 opened the pull request.

Commit:
commit#1

Commit:
commit#1

cosminflorin1 committed:
Add account done (-validation, -encryption)

cosminflorin1 committed:
DateOfBirth is now a Date

cosminflorin1 committed:
Added encryption -> cyclic dependency

cosminflorin1 committed:
Added encryption

cosminflorin1 committed:
Added encryption

cosminflorin1 committed:
Set username as id for accounts

AnaSireag committed:
[ACCOUNT]: delete account

Commit:
Add account validations

AnaDrambarean committed:
view account details

cosminflorin1 committed:
added get all method

AnaDrambarean committed:
AccountDetails class

Commit:
Account validations

Tarciziu reviewed (commented):
Recomand să ai o clasă separată Validator sau o alternativă ar mai fi să faci validări din adnotări
Exemplu pentru:
 -> validare mail:
@ Pattern(regexp = "^\\\w+([-+.']\\\w+)\*@\\\w+([-.]\\\w+)\*\\\\.\\\w+([-.]\\\w+)*$", message = "Email invalid!")
-> validare nume:
@ NotBlank(message = "Name is mandatory")

vulturana1 commented:
> Recomand să ai o clasă separată Validator sau o alternativă ar mai fi să faci validări din adnotări
> Exemplu pentru:
> -> validare mail:
> @ Pattern(regexp = "^\\w+([-+.']\\w+)*@\\w+([-.]\\w+)*\\.\\w+([-.]\\w+)*$", message = "Email invalid!")
> -> validare nume:
> @ NotBlank(message = "Name is mandatory")

vulturana1 commented:
Ok, o sa fac atunci validările din adnotări

Commit:
validations with annotations

cosminflorin1 committed:
moved Exceptions directory, renamed the repository interface

cosminflorin1 committed:
Merge remote-tracking branch 'origin/anaVultur' into addAccount

# Conflicts:
#	src/main/java/com/bootcamp/demo/model/Account.java

AnaDrambarean committed:
viewAccountDetails

cosmin-195 committed:
renamed packages

cosmin-195 committed:
removed print

Commit:
Account model

cosmin-195 committed:
removed print

cosmin-195 committed:
made adding an account atomic
now also checks if another account with the same user exists

cosmin-195 committed:
add / getAll from Controller now return proper http responses

cosmin-195 committed:
addAccount changes:
- checks if there isn't already an account with the same username
- is now atomic
- returns an appropriate http status along with the result

Commit:
#ValidationService

Commit:
merge with addAccount branch
